src/hist/basehist.py

```diff
@@ -461,3 +461,26 @@ def plot_pie(
         import hist.plot
 
         return hist.plot.plot_pie(self, ax=ax, **kwargs)
+
+    def stack(self: T, *args: Union[int, str]) -> Any:  # todo: should return Stack here
+        """
+        Returns a stack from a normal histogram axes.
+        """
+        from .stacks import Stack
+
+        if len(args) == 0:
+            raise ValueError("There should be histograms or axes in stack")
+
+        if isinstance(args[0], int):
+            stack_args = [self.axes[i] for i in args]
+            return Stack(*stack_args)
```

Comment:
I think we should only support one arg, which is the axis to stack _over_. Multiple args would make a ND stack, which we don't want to support, at least for now.

The argument should be the axis to stack over, which should be expanded. For example:

```python
ax = 0
stack_histograms = {i:self[i, ...] for i in range(len(ax[0]))])
return Stack(*stack_histograms)
```

Though I think we also need to keep track of the bin this came from too.

Reply:
What's the `ax` in this example?

Reply:
In my case, it was the integer axis number to expand. Though if it was a string axis name, since that is supported natively in both `h[{...}]` and `h.axis[...]`, that should work too. I didn't expand the dict correctly above, though.

Reply:
Updated the example above - it now no longer "remembers" the cell name, though. Ideally, each histogram would also have at least "h.name" set to the original cell "name". Maybe:
```python
for name, h in zip(self.axis[ax], stack_histograms)
    h.name = name
```

Reply:
If `self.axes[i]` is a `StrCategory`, which is most common, then `h.name` will be the category that this histogram came from. If it's an integer, then name is an integer. If it's a Regular/Variable, then "name" is a tuple with two floats, which is weird for something called "name". But I think it's a reasonable tradeoff - having the value is still useful.

Reply:
Is there any difference between self.axis and self.axes?